lib/util.js

```diff
@@ -386,29 +388,61 @@ module.exports = {
+                       'DEP0045'),
+  isBuffer: deprecate(isBuffer,
+                      'The `util.isBuffer` API is deprecated. Please use `Buffer.isBuffer()` instead.',
+                      'DEP0046'),
   isDeepStrictEqual(a, b) {
     if (internalDeepEqual === undefined) {
       internalDeepEqual = require('internal/util/comparisons')
         .isDeepStrictEqual;
     }
     return internalDeepEqual(a, b);
   },
-  isNull,
-  isNullOrUndefined,
-  isNumber,
-  isString,
-  isSymbol,
-  isUndefined,
-  isRegExp: types.isRegExp,
-  isObject,
-  isDate: types.isDate,
-  isError,
-  isFunction,
-  isPrimitive,
-  log,
+  isNull: deprecate(isNull,
+                    'The `util.isNull` API is deprecated. Please use `arg === null` instead.',
+                    'DEP0050'),
+  isNullOrUndefined: deprecate(isNullOrUndefined,
+                               'The `util.isNullOrUndefined` API is deprecated. ' +
+                               'Please use `arg === null || arg === undefined` instead.',
+                               'DEP0051'),
+  isNumber: deprecate(isNumber,
+                      'The `util.isNumber` API is deprecated. Please use `typeof arg === "number"` instead.',
+                      'DEP0052'),
+  isString: deprecate(isString,
+                      'The `util.isString` API is deprecated.  Please use `typeof arg === "string"` instead.',
+                      'DEP0056'),
+  isSymbol: deprecate(isSymbol,
+                      'The `util.isSymbol` API is deprecated.  Please use `arg === "symbol"` instead.',
+                      'DEP0057'),
+  isUndefined: deprecate(isUndefined,
+                         'The `util.isUndefined` API is deprecated. Please use `arg === undefined` instead.',
+                         'DEP0058'),
+  isRegExp: deprecate(types.isRegExp,
+                      'The `util.isRegExp` API is deprecated. Please use `arg instanceof RegExp` instead.',
+                      'DEP0055'),
+  isObject: deprecate(isObject,
+                      'The `util.isObject` API is deprecated. Please use `typeof arg === "object"` instead.',
+                      'DEP0053'),
+  isDate: deprecate(types.isDate,
+                    'The `util.isDate` API is deprecated.  Please use `arg instanceof Date` instead.',
+                    'DEP0047'),
+  isError: deprecate(isError,
+                     'The `util.isError` API is deprecated.  Please use `arg instanceof Error` instead.',
+                     'DEP0048'),
+  isFunction: deprecate(isFunction,
+                        'The `util.isFunction` API is deprecated.  Please use `typeof arg === "function"` instead.',
+                        'DEP0049'),
+  isPrimitive: deprecate(isPrimitive,
+                         'The `util.isPrimitive` API is deprecated. ' +
+                         'Please use `typeof arg !== "object" && typeof arg !== "function"` instead.',
```

Comment:
```suggestion
                         'Please use `Object(arg) !== arg` instead.',
```

Reply:
the source code:
```
function isPrimitive(arg) {
  return arg === null ||
         (typeof arg !== 'object' && typeof arg !== 'function');
}
```
I'm think it's ok to suggest what it's actually doing

Reply:
Either way, the docs and the code need to agree, and they currently don’t.

Yes, it would be fine, except that if there’s a future language change that adds a new object typeof value, node can fix it whereas users are unlikely to. (also, the current runtime suggestion omits the null check, so it’ll be wrong)